src/cart/crt_hlc.c

```diff
@@ -96,9 +109,35 @@ int crt_hlc_get_msg(uint64_t msg, uint64_t *hlc_out, uint64_t *offset)
 	return 0;
 }
 
-uint64_t crt_hlc2sec(uint64_t hlc)
+uint64_t crt_hlc2nsec(uint64_t hlc)
 {
-	return (hlc & ~CRT_HLC_MASK) / NSEC_PER_SEC;
+	return hlc / CRT_HLC_NSEC;
+}
+
+uint64_t crt_hlc_from_nsec(uint64_t nsec)
+{
+	return nsec * CRT_HLC_NSEC;
+}
+
+uint64_t crt_hlc2unixnsec(uint64_t hlc)
+{
+	return hlc / CRT_HLC_NSEC + CRT_HLC_START * NSEC_PER_SEC;
+}
+
+int crt_hlc_from_unixnsec(uint64_t unixnsec, uint64_t *hlc)
+{
+	uint64_t start = CRT_HLC_START * NSEC_PER_SEC;
+
+	/*
+	 * If the time represented by unixnsec is before the time represented by
+	 * CRT_HLC_START, then the conversion is impossible.
+	 */
+	if (unixnsec < start)
+		return -DER_INVAL;
```

Comment:
why not return 0 on error and actual value on success? would make usage easier, and return value of 0 is impossible for valid inputs

Reply:
If the input corresponds to CRT_HLC_START_SEC, then the output is a valid 0.

Reply:
sure, but cant we impose that input has to be above start_sec time?the real clocks should all be now above jan 1 2021 00:00:00

Reply:
OK, changed.